File: LeetCode/0387_First_Unique_Character_in_a_String/solution.js
/**
 * @param {string} s
 * @return {number}
 */
var firstUniqChar = function(s) {
    let obj={}
    for(let i=0;i<s.length;i++){
        if(obj[s[i]] == undefined) obj[s[i]]=i;
        else obj[s[i]] = "dupli"
    }
    for(let char in obj){
        if(obj[char] !="dupli") return obj[char] 
    }
    return -1
};

File: LeetCode/0387_First_Unique_Character_in_a_String/solution.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        unordered_map <char, int> obj;
        int ans = 1e9;
        for (int i = 0; i < s.size(); i++) {
                if (obj.find(s[i]) == obj.end())  obj[s[i]] =i;
                else obj[s[i]] = -1;
        }
        for (const auto & p : obj) {
            if (p.second  != -1)
                ans = min(ans,p.second);
        }
        return (ans == 1e9)?-1:ans;
    }
};